datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/deser/DurationDeserializer.java

```diff
@@ -79,20 +97,36 @@ protected DurationDeserializer withLeniency(Boolean leniency) {
                     deser = deser.withLeniency(leniency);
                 }
             }
+            if (format.hasPattern()) {
+                deser = DurationUnitParser.from(format.getPattern())
```

Comment:
@obarcelonap Are you sure this preserves the `leniency` setting from above? Or should follow same convention as above, i.e.
`deser = deser.withPattern(..) `

Reply:
Yes it is, `withPattern` is referenced when there is a value using the `Optional` monad.
As you already pointed, the non functional approach, would be:
```
deser = deser.withPattern(DurationUnitParser.from(format.getPattern()).orElse(null))
```
Let me know if you prefer this second approach.

Reply:
> would it not be better to simply throw an exception to let user know that fractions are not allowed for anything but seconds?

Didn't even think of that, was assuming no fractions. So yes, my vote would be "simpler is better".